tapdb/sqlc/migrations/000020_asset_unique_key.up.sql

```diff
@@ -1,4 +1,107 @@
+-- Step 1: If the assets were spent, some of the duplicates might not have been
+-- updated on that flag. To make sure we can properly group on the spent flag
+-- below, we now update all assets that are spent.
+UPDATE assets
+SET spent = true
+WHERE asset_id IN (SELECT a.asset_id
+                   FROM assets a
+                            JOIN managed_utxos mu
+                                 ON a.anchor_utxo_id = mu.utxo_id
+                            JOIN chain_txns ct
+                                 ON mu.txn_id = ct.txn_id
+                            LEFT JOIN asset_transfer_inputs ati
+                                      ON ati.anchor_point = mu.outpoint
+                   WHERE a.spent = false
+                     AND ati.input_id IS NOT NULL);
```

Comment:
Please correct me if my understanding is wrong: if `ati.input_id IS NOT NULL` then the asset is spent? And those are the rows that we ensure are set to spent, is that correct?

Reply:
Correct. We `LEFT JOIN` here, meaning that any field on `ati` is `NULL` if there is no matching spending transaction referencing the asset's on-chain outpoint.